test/util/fake-xml-http-request-test.js

```diff
@@ -1106,13 +1106,80 @@ if (typeof window !== "undefined") {
             assertEventOrdering("abort", 0, function (xhr) {
                 xhr.abort();
             });
+        });
 
-            assertEventOrdering("error", 0, function (xhr) {
-                xhr.respond(500, {}, "");
+        describe(".error", function () {
+            beforeEach(function () {
+                this.xhr = new sinon.FakeXMLHttpRequest();
             });
 
-            assertEventOrdering("load", 100, function (xhr) {
-                xhr.respond(200, {}, "");
+            it("sets response to empty string", function () {
+                this.xhr.response = "Partial data";
+
+                this.xhr.error();
+
+                assert.same(this.xhr.response, "");
+            });
+
+            it("sets responseText to empty string", function () {
+                this.xhr.responseText = "Partial data";
+
+                this.xhr.error();
+
+                assert.same(this.xhr.responseText, "");
+            });
+
+            it("sets errorFlag to true", function () {
```

Comment:
Set `errorFlag` to `false` to confirm that it changed. In theory `this.xhr.errorFlag` could be `true` before `.error()` is called in this test.